### ISSUE TITLE
Similarity maps with the new drawing code produce cropped and pixelated images

### ISSUE THREAD

**Issue by eguidotti** [rdkit/rdkit#6190]
**Describe the bug**
I'm trying to generate similarity maps with the new drawing code as described [here](https://greglandrum.github.io/rdkit-blog/posts/2020-01-03-similarity-maps-with-new-drawing-code.html). But images are cropped and pixelated.

**To Reproduce**
```python
from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem.Draw import SimilarityMaps
import io
from PIL import Image

def show_png(data):
    bio = io.BytesIO(data)
    img = Image.open(bio)
    return img
```

This is what I get with the new code:
```python
mol = Chem.MolFromSmiles('N=C(N)NCCc1c[nH]cn1')
d = Draw.MolDraw2DCairo(512, 512)
SimilarityMaps.GetSimilarityMapFromWeights(mol, [1 for i in range(mol.GetNumAtoms())], draw2d=d)
d.FinishDrawing()
show_png(d.GetDrawingText())
```
<img width="514" alt="Screenshot 2023-03-12 at 10 45 48" src="https://user-images.githubusercontent.com/24247667/224536884-66667f60-ed5e-4763-aec1-8509e450759a.png">

This is what I get with the old code:
```python
SimilarityMaps.GetSimilarityMapFromWeights(mol, [1 for i in range(mol.GetNumAtoms())])
```
<img width="647" alt="Screenshot 2023-03-12 at 10 46 02" src="https://user-images.githubusercontent.com/24247667/224536894-99346269-972c-4311-8294-030eb3f1b525.png">

**Expected behavior**
I would expect images that are not cropped or pixelated, as in the old code and in the [blog post](https://greglandrum.github.io/rdkit-blog/posts/2020-01-03-similarity-maps-with-new-drawing-code.html).

**Screenshots**
See above.

**Configuration (please complete the following information):**
 - RDKit version: 2022.09.5
 - OS: MacOS Monterey
 - Python version (if relevant): 3.9
 - Are you using conda? no
 - If you are using conda, which channel did you install the rdkit from? pypi
 - If you are not using conda: how did you install the RDKit? pip install rdkit-pypi

**Comment by iaacornus** [comment]
hi. i've been experiencing the same issue and never found a resolve all across the internet, perhaps there's some workaround for this one?

**Comment by ChemMatt** [comment]
I had the same problem. Looks like it was fixed in #8335 and is present in release [2025_03_1 (Q1 2025) Release](https://github.com/rdkit/rdkit/releases/tag/Release_2025_03_1). Now allows for an option to decrease grid size as below.

Warm regards,
Matt

example:
```
SimilarityMaps.GetSimilarityMapFromWeights(mol, weights, draw2d, gridResolution=0.01)
```

Old (gridResolution=0.1):
![Image](https://github.com/user-attachments/assets/8525d926-7edc-4926-a423-58b87a27360a)
New (gridResolution=0.01):
![Image](https://github.com/user-attachments/assets/442267ef-d42d-4ac0-88db-2a0718aebbc2)
